### PR TITLE
Sourcetree redux

### PR DESCRIPTION
**Pull request by bomsy**
Associated Issue: #3260

### Summary of Changes

* Add source-tree reducer
* Add source-tree actions
* On mount of Managed Tree if there is an expanded cache we load that

Example test plan:

- [x] Load debugger
- [x] Load debuggee `http://firefox-dev.tools/debugger-examples/examples/todomvc/`
- [x] Click the source tree
- [x] Adjust to vertical or horizontal layout

### CODE REVIEW

**Comment by jasonLaster** [comment]
This looks really good! 👍

File: src/reducers/tests/source-tree.spec.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    const state = InitialState();
+    const expanded = new Set(["foo", "bar"]);
+    const updatedState = update(state, {
+      type: "SET_EXPANDED_STATE",
+      expanded
+    });
+
+    expect(updatedState.expanded).toBe(expanded);
+  });
+});
```
Comment (jasonLaster) [inline src/reducers/tests/source-tree.spec.js]: i think this test is redundant. we get the same coverage from the action test.

Comment (bomsy) [reply src/reducers/tests/source-tree.spec.js]: lol .. yeah ... tot so too. never written a reducer test, got excited with the tests :)

Comment (bomsy) [reply src/reducers/tests/source-tree.spec.js]: will pull it out.

**Comment by jasonLaster** [comment]
it looks like we dispatch two actions per toggle:



Is it possible to avoid?

**Comment by bomsy** [comment]
@jasonLaster i can't seem to replicate this. How are you logging the expanded state?

**Comment by bomsy** [comment]
i see this

**Pull request merged**